File: Android/app/src/main/java/com/edu/primary/utils/DatabaseInitializer.java
package com.edu.primary.utils;

import android.content.Context;
import com.edu.primary.BuildConfig;
import com.edu.primary.constants.AppConstants;
import com.edu.primary.database.AppDatabase;
import com.edu.primary.database.entity.QuestionEntity;
import io.reactivex.Completable;
import io.reactivex.schedulers.Schedulers;

import java.util.List;

public class DatabaseInitializer {
    // 根据BuildConfig.DEBUG决定题目数量：debug包5道，release包40道
    private static final int QUESTIONS_PER_GRADE = BuildConfig.DEBUG ? AppConstants.DEBUG_QUESTIONS_PER_PRACTICE : AppConstants.RELEASE_QUESTIONS_PER_PRACTICE;

    /**
     * 初始化数学题目数据
     * 如果数据库中没有数学题目，则生成随机题目
     * 使用RxJava替代new Thread，提高代码一致性
     */
    public static Completable initializeMathQuestions(Context context) {
        return Completable.fromAction(() -> {
            // 使用ApplicationContext防止内存泄漏
            AppDatabase database = AppDatabase.getInstance(context.getApplicationContext());
            
            // 检查是否已有数学题目
            int totalCount = 0;
            for (int grade = AppConstants.MIN_GRADE; grade <= AppConstants.MAX_GRADE; grade++) {
                totalCount += database.questionDao().getQuestionCount(
                    AppConstants.SUBJECT_MATH, grade);
            }
            
            // 如果没有题目，则生成
            if (totalCount == 0) {
                List<QuestionEntity> questions = QuestionDataGenerator.generateAllMathQuestions(
                    QUESTIONS_PER_GRADE);
                database.questionDao().insertQuestions(questions);
            }
        }).subscribeOn(Schedulers.io());
    }

    /**
     * 为指定年级生成数学题目
     * 使用RxJava替代new Thread，提高代码一致性
     */
    public static Completable generateMathQuestionsForGrade(Context context, int grade, int count) {
        return Completable.fromAction(() -> {
            // 使用ApplicationContext防止内存泄漏
            AppDatabase database = AppDatabase.getInstance(context.getApplicationContext());
            List<QuestionEntity> questions = QuestionDataGenerator.generateMathQuestions(grade, count);
            database.questionDao().insertQuestions(questions);
        }).subscribeOn(Schedulers.io());
    }
}

File: Android-1/app/src/main/java/com/edu/primary/util/DatabaseInitializer.kt
package com.edu.primary.util

import android.content.Context
import com.edu.primary.data.local.AppDatabase
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

/**
 * 数据库初始化：若数学题表为空，则调用 [QuestionDataGenerator] 生成并插入各年级题目。
 * 题目数量由 DEBUG/RELEASE 与 [AppConstants] 决定。
 */
object DatabaseInitializer {
    private val questionsPerGrade
        get() = if (com.edu.primary.BuildConfig.DEBUG)
            AppConstants.DEBUG_QUESTIONS_PER_PRACTICE
        else
            AppConstants.RELEASE_QUESTIONS_PER_PRACTICE

    /** 若各年级数学题总数为 0，则生成并插入默认数学题。 */
    suspend fun initializeMathQuestions(context: Context) = withContext(Dispatchers.IO) {
        val db = AppDatabase.getInstance(context)
        val total = (AppConstants.MIN_GRADE..AppConstants.MAX_GRADE).sumOf { grade ->
            db.questionDao().getCount(AppConstants.SUBJECT_MATH, grade)
        }
        if (total == 0) {
            val list = QuestionDataGenerator.generateAllMathQuestions(questionsPerGrade)
            db.questionDao().insertAll(list)
        }
    }

    /** 为指定年级生成并插入 [count] 道数学题。 */
    suspend fun generateMathForGrade(context: Context, grade: Int, count: Int) = withContext(Dispatchers.IO) {
        val db = AppDatabase.getInstance(context)
        val list = QuestionDataGenerator.generateMathQuestions(grade, count)
        db.questionDao().insertAll(list)
    }
}
